src/pykeen/trackers/tensorboard.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+
+"""An adapter for TensorBoard."""
+
+import pathlib
+import time
+from typing import Any, Dict, Mapping, Optional, TYPE_CHECKING, Type
+
+from .base import ResultTracker
+from ..constants import PYKEEN_LOGS
+from ..utils import flatten_dictionary
+
+if TYPE_CHECKING:
+    import torch.utils.tensorboard
+
+__all__ = [
+    'TensorBoardResultTracker',
+]
+
+
+class TensorBoardResultTracker(ResultTracker):
+    """A tracker for TensorBoard."""
+
+    #: The class that's used to instantiate a summarywriter
+    summary_writer_cls: Type['torch.utils.tensorboard.SummaryWriter']
+
+    def __init__(
+        self,
+        experiment_path: Optional[str] = None,
+        experiment_name: Optional[str] = None,
+        tags: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        Initialize result tracking via Tensorboard.
+
+        :param experiment_path:
+            The experiment path. A custom path at which the tensorboard logs will be saved.
+        :param experiment_name:
+            The name of the experiment, will be used as a sub directory name for the logging. If no default is given,
+            the current time is used. If set, experiment_path is set, this argument has no effect.
+        :param tags:
+            The additional run details which are presented as tags to be logged
+        """
+        import torch.utils.tensorboard
+        self.summary_writer_cls = torch.utils.tensorboard.SummaryWriter
+        self.tags = tags
+
+        if experiment_path is None:
+            if experiment_name is None:
+                experiment_name = time.strftime('%Y-%m-%d-%H-%M-%S')
+            path = PYKEEN_LOGS.joinpath("tensorboard", experiment_name)
+        elif isinstance(experiment_path, str):
+            path = pathlib.Path(experiment_path)
+        self.path = path
+
+    def start_run(self, run_name: Optional[str] = None) -> None:  # noqa: D102
+        self.writer = self.summary_writer_cls(log_dir=self.path, comment=run_name)
```

Comment:
The documentation for SummarWriter says the following about the `comment` argument:
```
comment (string): Comment log_dir suffix appended to the default
              ``log_dir``. If ``log_dir`` is assigned, this argument has no effect.
```

Is this actually doing anything here?

Reply:
Also I think the writer should be assigned in the `__init__``

Reply:
Thanks - I have corrected both now, although there is now no start_run overwritten in this class....not sure if that is normal for these trackers.

Reply:
Sorry I missed this comment before. It's okay if you don't use part of the functionality